src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -362,14 +427,38 @@ public enum RisingRivals implements LogicCardInfo {
           pokeBody "Boundary Aura", {
             text "Apply Weakness for each Pokémon as ×2 instead."
             delayedA {
+              def flag = false
+              before PROCESS_ATTACK_EFFECTS, {
+                if (ef.attacker == self){
+                  bg.dm().each{
+                    if (it.to.owner != self.owner && it.to.active && !it.flags.contains(DamageManager.DamageFlag.NO_WEAKNESS)) {
+                      it.flags.add(DamageManager.DamageFlag.NO_WEAKNESS)
+                      flag = true
+                    }
+                  }
+                }
+              }
+              after PROCESS_ATTACK_EFFECTS, {
+                if(ef.attacker){
+                  bg.dm().each {
+                    if (it.from.active && it.to.active && it.dmg.value && flag) {
+                      for(Type t1:Type.values()){//Doing it this way in case of 4x weakness
+                        if(it.from.types.contains(t1) && it.to.weaknesses.find{it.type == t1}){
+                          it.dmg *= 2
+                        }
+                      }
+                    }
+                  }
+                }
+              }
```

Comment:
Use a GET_WEAKNESS getter for this. Check BaseSetNG PORYGON to see how you can create a new object with the X2 feature but the same weakness.